File: LeetCode/1290. Convert Binary Number in a Linked List to Integer/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

/**
 * Runtime: 0 ms
 * Memory Usage: 38.9 MB 
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        int ans = 0;
        for (ListNode p = head; p != null; p = p.next) {
            ans = (ans << 1) | p.val;
        }
        return ans;
    }
}

File: LeetCode/1290. Convert Binary Number in a Linked List to Integer/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

/**
 * Runtime: 0 ms
 * Memory Usage: 8 MB 
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        int ans = 0;
        for (auto p = head; p != nullptr; p = p->next) {
            ans = ans << 1 | p->val;
        }
        return ans;
    }
};
